app/assets/javascripts/discourse/app/controllers/tag-show.js

```diff
@@ -91,9 +92,45 @@ export default DiscoverySortableController.extend(
       return this._isFilterPage(filter, "unread") && topicsLength > 0;
     },
 
-    @discourseComputed("list.filter", "list.topics.length")
-    showResetNew(filter, topicsLength) {
-      return this._isFilterPage(filter, "new") && topicsLength > 0;
+    @discourseComputed("list.filter")
+    new(filter) {
+      return this._isFilterPage(filter, "new");
+    },
+
+    @discourseComputed("new")
+    showTopicsAndRepliesToggle(isNew) {
+      return isNew && this.currentUser?.new_new_view_enabled;
```

Comment:
This logic is being duplicated in quite a number of places. Is there a way we can avoid this?

Reply:
Are you referring to the duplicate `showTopicsAndRepliesToggle` property in [`tag-show.js`](https://github.com/discourse/discourse/blob/8b3eca056bc9f10224ddebc6ac1f63c69b3ce2ba/app/assets/javascripts/discourse/app/controllers/tag-show.js) and [`discovery/topics.js`](https://github.com/discourse/discourse/blob/8b3eca056bc9f10224ddebc6ac1f63c69b3ce2ba/app/assets/javascripts/discourse/app/controllers/discovery/topics.js)? I could extract the logic of the property into a function and then export/import/call that function in these 2 files, but they are already almost complete duplicates of each other, so I don't know if it makes sense to DRY just one property/line when there's already so much duplicate logic.